shell/platform/darwin/macos/framework/Source/FlutterEngine.mm

```diff
@@ -181,6 +200,12 @@ - (instancetype)initWithName:(NSString*)labelPrefix
   _textures = [[NSMutableDictionary alloc] init];
   _allowHeadlessExecution = allowHeadlessExecution;
 
+  NSNotificationCenter* notificationCenter = [NSNotificationCenter defaultCenter];
```

Comment:
The observer needs to be removed in the notification center explicitly in the `dealloc`. Not doing so will lead to a dangling pointer on older versions of macOS.

Reply:
The discussion in https://developer.apple.com/documentation/foundation/nsnotificationcenter/1415360-addobserver says macOS 10.11. I am not sure we support stuff that old but its good practice to cleanup anyway.

Reply:
10.11 is our lower limit, which is why I didn't unregister it.

> but its good practice to cleanup anyway

My experience with modern ObjC (in iOS projects that target 9+) is that it's become standard practice not to unregister, given that it's explicitly documented that you no longer need to. As with the transition from manual ref counting to ARC, it's one of those things that looks really wrong at first, but you get used to :)